### PR TITLE
Support "Edit attachment" action for featured file attachments

### PR DESCRIPTION
**Pull request by leenagupte**
Trello: https://trello.com/c/lZKUDZbj

# What's changed?

Adds an "Edit attachment" link to the featured attachment preview block on the attachments index page. Clicking on "Edit attachment" takes the user to existing edit page where they can edit the title or replace the file.

# Expected changes
Screenshot 2020-03-12 at 16 27 02

Screenshot 2020-03-13 at 11 23 09

# Note to reviewer
These changes ignore the new designs for the edit attachment page. There is follow-up work to make this consistent between the modal for inline attachments and the page for featured attachments.

### CODE REVIEW

**Comment by leenagupte** [comment]
@benthorner I've made a stab at the suggested changes and fixed up.

File: app/views/file_attachments/edit.html.erb [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
+                    end %>
+
+<% content_for :back_link, render_back_link(href: back_link_path,
                                             data_attributes: { "modal-action": "back" }) %>
 
 <div class="govuk-grid-row">
-  <div class="govuk-grid-column-full">
+  <div class="<%= rendering_context == 'modal' ? 'govuk-grid-column-full' : 'govuk-grid-column-two-thirds' %>">
+    <% data = {}
+       data["modal-action"] == "update" if rendering_context == "modal" %>
```
Comment (benthorner) [inline app/views/file_attachments/edit.html.erb]: (I think we can do the same here, as it's only used once, in a modal context.)

Comment (leenagupte) [reply app/views/file_attachments/edit.html.erb]: @benthorner Do the same here as?

Comment (benthorner) [reply app/views/file_attachments/edit.html.erb]: Ah, I was referring to [this comment, a few lines above](https://github.com/alphagov/content-publisher/pull/1901#discussion_r392351267). I'll remember to be more precise next time 👌.

Comment (leenagupte) [reply app/views/file_attachments/edit.html.erb]: Yeah. I think those changes are in a different commit, so they didn't appear together.

Comment (leenagupte) [reply app/views/file_attachments/edit.html.erb]: I'll remove this commit and rebase.

**Comment by leenagupte** [comment]
@benthorner This ready for re-review.

**Pull request merged**